proto/src/prost/cosmos.staking.v1beta1.rs

```diff
@@ -825,301 +857,411 @@ pub mod query_client {
+    /// last_total_power tracks the total amounts of bonded tokens recorded during
+    /// the previous end block.
+    #[prost(bytes="vec", tag="2")]
+    pub last_total_power: ::prost::alloc::vec::Vec<u8>,
+    /// last_validator_powers is a special index that provides a historical list
+    /// of the last-block's bonded validators.
+    #[prost(message, repeated, tag="3")]
+    pub last_validator_powers: ::prost::alloc::vec::Vec<LastValidatorPower>,
+    /// delegations defines the validator set at genesis.
+    #[prost(message, repeated, tag="4")]
+    pub validators: ::prost::alloc::vec::Vec<Validator>,
+    /// delegations defines the delegations active at genesis.
+    #[prost(message, repeated, tag="5")]
+    pub delegations: ::prost::alloc::vec::Vec<Delegation>,
+    /// unbonding_delegations defines the unbonding delegations active at genesis.
+    #[prost(message, repeated, tag="6")]
+    pub unbonding_delegations: ::prost::alloc::vec::Vec<UnbondingDelegation>,
+    /// redelegations defines the redelegations active at genesis.
+    #[prost(message, repeated, tag="7")]
+    pub redelegations: ::prost::alloc::vec::Vec<Redelegation>,
+    #[prost(bool, tag="8")]
+    pub exported: bool,
+}
+/// LastValidatorPower required for validator set update logic.
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct LastValidatorPower {
+    /// address is the address of the validator.
+    #[prost(string, tag="1")]
+    pub address: ::prost::alloc::string::String,
+    /// power defines the power of the validator.
+    #[prost(int64, tag="2")]
+    pub power: i64,
+}
 /// StakeAuthorization defines authorization for delegate/undelegate/redelegate.
 ///
 /// Since: cosmos-sdk 0.43
 #[derive(Clone, PartialEq, ::prost::Message)]
 pub struct StakeAuthorization {
     /// max_tokens specifies the maximum amount of tokens can be delegate to a validator. If it is
     /// empty, there is no spend limit and any amount of coins can be delegated.
-    #[prost(message, optional, tag = "1")]
+    #[prost(message, optional, tag="1")]
     pub max_tokens: ::core::option::Option<super::super::base::v1beta1::Coin>,
     /// authorization_type defines one of AuthorizationType.
-    #[prost(enumeration = "AuthorizationType", tag = "4")]
+    #[prost(enumeration="AuthorizationType", tag="4")]
     pub authorization_type: i32,
     /// validators is the oneof that represents either allow_list or deny_list
-    #[prost(oneof = "stake_authorization::Validators", tags = "2, 3")]
+    #[prost(oneof="stake_authorization::Validators", tags="2, 3")]
     pub validators: ::core::option::Option<stake_authorization::Validators>,
 }
 /// Nested message and enum types in `StakeAuthorization`.
 pub mod stake_authorization {
     /// Validators defines list of validator addresses.
     #[derive(Clone, PartialEq, ::prost::Message)]
-    pub struct ValidatorVec {
-        #[prost(string, repeated, tag = "1")]
+    pub struct Validators {
```

Comment:
This is one of the breaking changes. In the proto file, a `oneof` field is named `validators` and there is a message type `Validators` defined alongside it. It worked in prost 0.6 because for some reason, the type name for the message struct was mangled.

Reply:
This is due to https://github.com/tokio-rs/prost/issues/505. And I have just learned that generate-and-hack was our way to deal with it earlier as well.